.github/workflows/codegen.yml

```diff
@@ -0,0 +1,42 @@
+# All code generation should be run prior to pull request. Running it again should not produce a diff.
+name: "Codegen Verifier"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'ccip-develop'
+
+jobs:
+  codegen-verifier:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        go-version: ['1.21']
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Install Mockery
+        run: go install github.com/vektra/mockery/v2@v2.43.0
+      - name: Display Go version
+        run: go version
+      - name: Generate mocks
+        run: make generate
+      - name: test git
+        run: git status
+      - name: ensure no changes
```

Comment:
nice to have/ We could even include a go mod tidy and check that we have no difference.

Reply:
Nice :+1: